[Chan Qin Liang] iP

qlchan24 opened the pull request.

Commit:
Add Gradle support

qlchan24 committed:
Implement skeletal version of Duke

The chatbot has a proper layout and can greet the user, echo a command,
and exit upon command.

qlchan24 committed:
Add basic functionality of a task list

Allow for adding tasks to a list, and upon command, display all items
in the list so far.

qlchan24 committed:
Use a class to represent tasks

Replaced the ArrayList of Strings with an array of Tasks to conform to
object-oriented design principles.

qlchan24 committed:
Add the ability to mark tasks as done

Implement new functions to mark as done and update design and messages
 to support the new functionality.

qlchan24 committed:
Use Inheritance to support multiple task types

There are similarities between the task types hence inheritance will
help reduce code duplication and also allow for polymorphism when
handling tasks.

Implement classes Todo, Deadline and Event that extends the Task
class.

qlchan24 committed:
Reassign formatting of tasks and messages

Tasks are formatted primarily by their own classes and other messages
are formatted by the appropriate functions.

qlchan24 committed:
Allow users to create tasks from any of the task subtypes.

Modify the input processing to look for specific keywords representing
 the subtype of Task to create, and add new messages to complement the
  creation of these tasks.

qlchan24 committed:
Add automated Text UI Testing

Using the I/O redirection technique, update the files in the
text-ui-test folder to compare intended output with actual output.

qlchan24 committed:
Add error handling

Output the appropriate message when the user input is incorrect.

qlchan24 committed:
Use exceptions to handle errors

Where appropriate, create new exception classes. Throw exceptions and
catch them when suitable.

qlchan24 committed:
Use ArrayList to store tasks rather than array

The use of Arraylist eliminates the need to keep track of a count field
and an id field and eliminates the need for remapping if a task is
removed.

Modify all the functions to accept an arraylist instead of an array, and
 remove any redundant fields for the task class.

qlchan24 committed:
Show a message if the user has already done a task

Users should not be allowed to mark a task as done if they have already
done so for the same task.

qlchan24 committed:
Add functionality for deleting tasks

Users should have a way to clear tasks that have already been done or
cancelled or wrongly saved.

Create new functions and messages to delete any task from the existing
list.

qlchan24 committed:
Add enum for task types

Enum will ensure that the values for task type will be restricted to the
 types of tasks that are actually supported.

qlchan24 committed:
Add ability to save to disk

The user will likely want to retrieve the tasks he/she saved in the
previous session.

Hence, save the tasks of the current session to the user's system disk
and retrieve it on every initialisation.

qlchan24 committed:
Incorporate Java date and time data types

Existing tasks saves the date/time as strings which provides very little
 flexibility when converting to another format or performing operations
 on the date/time.

 Use specific date/time types to allow for further manipulation of the
 date/time than the existing strings.

qlchan24 committed:
Merge branch 'branch-Level-7'

qlchan24 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

qlchan24 committed:
Refactor the code to extract out related code as classes

Existing code has most of the primary functions and fields in a single
class. This does not conform to OOP principles.

Move the related members into classes of their own.

qlchan24 committed:
Add unit tests for TaskList and Parser

Allow for more comprehensive testing by creating test classes and
overriding equals() and hashcode() for the classes used in the tests to
facilitate the assertions.

qlchan24 committed:
Add Javadoc to most non-private classes and methods.

Provide documentation of most of the existing code.

qlchan24 committed:
Tweak the code ensure compliance with coding standard

qlchan24 committed:
Add find command

Allow users to search for tasks in their list using a search query.

Implement the relevant ui and tasklist functions together with a new
command type.

qlchan24 committed:
Merge branch 'branch-A-JavaDoc'

qlchan24 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/AddCommand.java
#	src/main/java/Deadline.java
#	src/main/java/DeleteCommand.java
#	src/main/java/DoneCommand.java
#	src/main/java/Event.java
#	src/main/java/Storage.java
#	src/main/java/Task.java

qlchan24 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/ListCommand.java
#	src/main/java/Ui.java

qlchan24 committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

qlchan24 committed:
Integrate Gradle into project

qlchan24 committed:
Correct checkstyle violations

qlchan24 committed:
Resolve storage bug

Saving to storage previously included bugs due to writing without truncating.

qlchan24 committed:
Change logo

The previous logo used Unicode characters which are able show up correctly when building and running with the IDE or shell scripts.

However, now that Gradle has been integrated into the deployment, there are some characters that cannot show up properly even after explicitly defining the compiler and the VM file/console encoding in the gradle build/properties files, and a fix has not yet been found.

Hence use characters that will show up correctly when deployed and run with gradle.

qlchan24 committed:
Create GUI

Previous application was run from the shell. However, there was limited flexibility in design for such applications and does not look as aesthetically pleasing.

Implement a GUI to give users visual feedback while still retaining the practicality of a CLI application.

qlchan24 committed:
Merge branch 'branch-Level-10'

fyshhh commented on src/main/java/DeleteCommand.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws IOException, IndexOutOfBoundsException {
+        Task current=tasks.getTaskAtIndex(id);
```
i believe operators should be surrounded by a space character!

fyshhh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.format.DateTimeParseException;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.Region;
+import javafx.stage.Stage;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
```
perhaps consider separating the java and javafx imports?

fyshhh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,69 @@
+            return Ui.MESSAGE_TASK_ID_MISSING;
+        }
+        try {
+            String response=c.execute(tasks, ui, storage);
```
similarly, add some space characters!

fyshhh reviewed (commented):
Good work overall! Small changes suggested to adhere to coding standards. I love the Mario theme!

qlchan24 committed:
Add assertions to code

Ensure that important assumptions hold during the running of the program.

galvinleow commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,101 @@
+     *
+     * @return List of attributes of the deadline.
+     */
+    @Override
+    public String[] attributeList() {
+        return new String[] { "D", getName(), dueDate.toString(), time.toString(), String.valueOf(isDone()) };
+    }
```
maybe can rename it, like initAttributeList() or something

galvinleow commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,100 @@
+     * @return List of attributes of the event.
+     */
+    @Override
+    public String[] attributeList() {
```
same as above, more suggestion like createAttributeList(), not sure of the functionality of this method.

galvinleow commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,86 @@
+     * @param tasks List of tasks.
+     * @throws IOException If an error occurred when saving to disk.
+     */
+    void updateMemory(ArrayList<Task> tasks) throws IOException {
```
maybe don't use the default accessor type

galvinleow commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,86 @@
+     * @throws BlankTaskException If the task provided in the list has a blank name.
+     * @throws IOException        If there was an error in saving to disk.
+     */
+    ArrayList<Task> load() throws BlankTaskException, IOException {
```
same as above, maybe dont use default accessor type

galvinleow commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,85 @@
+public class Ui {
+
+    // Messages used in the package classes
+    static final String MESSAGE_EXIT = "Hey! Come back here! You big-a monkey!";
+    static final String MESSAGE_INVALID_ID =
+            "That task wasn't even on the list! You can save the princess if you're that free...";
+    static final String MESSAGE_TASK_ID_MISSING = "You didn't give me the task number to work with...";
+    static final String MESSAGE_WRONG_FORMAT = "The date format should be in DD-MM-YY and the time in HH:MM";
+    static final String MESSAGE_ALR_DONE = "Do you happen to have short term memory?";
+    static final String MESSAGE_ERROR_IO = "An error occurred while loading/saving to disk";
```
same as above, try not to use default accessor type

galvinleow reviewed (commented):
In my opinion, do not use the default accessor type.

qlchan24 committed:
Refactor and clean code

Improve the general code quality of the codebase.

qlchan24 committed:
Merge pull request #2 from qlchan24/branch-A-CodeQuality

Refactor and clean code

qlchan24 committed:
Merge branch 'master' of https://github.com/qlchan24/ip

qlchan24 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Storage.java
#	src/main/java/Ui.java

qlchan24 committed:
Merge pull request #1 from qlchan24/branch-A-Assertions

Add assertions to code

qlchan24 committed:
Merge branch 'master' of https://github.com/qlchan24/ip

qlchan24 committed:
Allow users to define custom alias and provide some preset alias

Advanced users may find it more efficient to use shorter keywords for the respective commands.

qlchan24 committed:
Validate and restrict illegal custom alias

Users are able to map a default command to another default command, effectively preventing further use of the latter, which may break the program.

Hence users should be stopped from creating such alias.

qlchan24 committed:
Add GUI enhancements

Modify the style of the GUI to look more appealing.

qlchan24 committed:
Add more unit tests

qlchan24 committed:
Change instances named after 'Duke' to 'Mario'

qlchan24 committed:
Expand GUI window and set its title

qlchan24 committed:
Update readme and screenshot of application

qlchan24 committed:
Set theme jekyll-theme-merlot

qlchan24 committed:
Update User Guide

qlchan24 committed:
Merge remote-tracking branch 'origin/master'

qlchan24 committed:
Set theme jekyll-theme-cayman

qlchan24 committed:
Fix bug as a result of invalid user input to 2 commands

Commands 'done' and 'delete' that receives numerical arguments will result in the GUI not giving any feedback when non-numeric input is given.

Catch the exception and show a message on the GUI instead.

qlchan24 committed:
Merge remote-tracking branch 'origin/master'
